### PR TITLE
Fixes #10431: add UI for bulk content host subscriptions.

### PR DESCRIPTION
**Pull request by waldenraines**
http://projects.theforeman.org/issues/10431

### CODE REVIEW

**Comment by waldenraines** [comment]
Note that this requires https://github.com/Katello/katello/pull/6241

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
[test]

**Comment by johnpmitsch** [comment]
@waldenraines I was expecting the Add and the Remove to be two separate tabs. Since all the subscriptions are in one page, it looks like their totals are for the total subscriptions used. ex. (2 out 10) When I select only one or two content hosts, I am still seeing the totals for all the subscriptions. This makes it hard to know what subscriptions are possible to remove from the content hosts selected. Is there a reason not to have two tabs (add and list/remove) where the add tab shows global subscription counts and the list/remove tabs shows the count for the specified content hosts?

**Comment by waldenraines** [comment]
> @waldenraines I was expecting the Add and the Remove to be two separate tabs. Since all the subscriptions are in one page, it looks like their totals are for the total subscriptions used. ex. (2 out 10) When I select only one or two content hosts, I am still seeing the totals for all the subscriptions. This makes it hard to know what subscriptions are possible to remove from the content hosts selected.

Yes, that's true.  Would it be helpful if I removed those totals?  @thomasmckay?

> Is there a reason not to have two tabs (add and list/remove) where the add tab shows global subscription counts and the list/remove tabs shows the count for the specified content hosts?

If you put the subscriptions in two tabs and then select one host with the subscription and one without what would you show on the add/remove tabs?  

In  my mind this UI is for the case where you just want to select a bunch of hosts and add or remove a subscription and you don't care if they host already has it or not.  It's sort of a brute force method.

**Comment by johnpmitsch** [comment]
>In my mind this UI is for the case where you just want to select a bunch of hosts and add or remove a
>subscription and you don't care if they host already has it or not. It's sort of a brute force method.

@waldenraines that makes sense. I imagine you would still want the totals to see if you could apply X number of subscription foo to X hosts. Is there a way we can communicate to the user that it doesn't matter if the host already has the subscription when adding them? And also make sure they know nothing will be removed if the host doesn't have the subscription when removing a subscription? I found this a little confusing at first, though it seems intuitive now. Just a thought, others may disagree

**Comment by waldenraines** [comment]
> Is there a way we can communicate to the user that it doesn't matter if the host already has the subscription when adding them? And also make sure they know nothing will be removed if the host doesn't have the subscription when removing a subscription? I found this a little confusing at first, though it seems intuitive now. Just a thought, others may disagree

I just added a message, see what you think.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/bulk-actions-subscriptions.html [hunk 1]
```diff
@@ -1,40 +1,84 @@
 <span page-title>{{ 'Content Host Bulk Subscriptions' | translate }}</span>
 
-<div class="details">
-  <section>
-    <h4 translate>Content Host Subscriptions</h4>
+<div bst-alert="info">
+  <p translate>
+    This action will affect only those Content Hosts that require a change.
+    If the Content Host does not have the subscription it will not be removed, for example.
```
Comment (johnpmitsch) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/bulk-actions-subscriptions.html]: Looks good to me! If I am being super nitpicky I would rephrase the second sentence "For example, if the Content Host does not have the selected subscription, it will not be removed.", but I will leave that up to you.

**Comment by johnpmitsch** [comment]
@waldenraines @parthaa will take a closer look at this and the updates in https://github.com/Katello/katello/pull/6241 by the end of the week!

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/bulk-actions-subscriptions.html (hunk 1 above)
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/bulk-actions-subscriptions.html]: See what you think about my reword.

**Comment by johnpmitsch** [comment]
@waldenraines we should probably display the "No hosts selected error" to users when trying to attach without any hosts checked

{"displayMessage":"No hosts have been specified.","errors":["No hosts have been specified."]}

**Comment by johnpmitsch** [comment]
@waldenraines as well as displaying an error when having no subscriptions checked and trying to add or remove

{"displayMessage":"undefined method `inject' for nil:NilClass","errors":["undefined method `inject' for nil:NilClass"]}

**Comment by waldenraines** [comment]
> @waldenraines as well as displaying a custom error message when having no subscriptions checked and trying to add or remove
> 
> {"displayMessage":"undefined method inject' for nil:NilClass","errors":["undefined methodinject' for nil:NilClass"]}

Not displaying the error I will fix.  The actual error coming from the API should have been fixed in the API pull request.

**Comment by waldenraines** [comment]
@johnpmitsch updated.

**Comment by johnpmitsch** [comment]
> @waldenraines as well as displaying a custom error message when having no subscriptions checked and trying to add or remove
> 
> {"displayMessage":"undefined method inject' for nil:NilClass","errors":["undefined methodinject' for nil:NilClass"]}
> Not displaying the error I will fix. The actual error coming from the API should have been fixed in the API pull request.

@parthaa ^ My fault for not catching this during review of the API PR, can you this by displaying a proper error message with a follow-up PR?

**Comment by parthaa** [comment]
> @parthaa ^ My fault for not catching this during review of the API PR, can you fix this by displaying a proper error message with a follow-up PR?

@waldenraines can ya take care of this blooper :).

**Comment by waldenraines** [comment]
> @waldenraines can ya take care of this blooper :).

I request we do it as part of another PR as it's not part of the UI in any way whatsoever.

See http://projects.theforeman.org/issues/16369

**Comment by johnpmitsch** [comment]
ACK

**Comment by waldenraines** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by ehelms** [comment]
[test]
